bazel/grpc_build_system.bzl

```diff
@@ -71,17 +71,13 @@ def _update_visibility(visibility):
     # Visibility rules prefixed with '@grpc_' are used to flag different visibility rule
     # classes upstream.
     VISIBILITY_TARGETS = {
-        "alt_grpc_legacy": [],
-        "alt_grpc++_legacy": [],
-        "endpoint_tests": [],
         "grpc_opencensus_plugin": ["//visibility:public"],
-        "grpc_resolver_fake": [],
         "public": ["//visibility:public"],
     }
     final_visibility = []
     for rule in visibility:
         if rule.startswith("@grpc:"):
-            for replacement in VISIBILITY_TARGETS[rule[6:]]:
+            for replacement in VISIBILITY_TARGETS.get(rule[6:], []):
```

Comment:
Nit: Seems like it could be frustrating if you fat fingered a visibility target. Maybe log a warning in case `rule[6:] not in VISIBILITY_TARGETS`?

Reply:
I changed it back... here I think you're right.
The internal version needs to accept anything to allow imports to come unimpeded, and I thought to match that here, but I don't actually need to and the strictness is good here.